### PR TITLE
roslisp-msg-protocol: looking up symbols from constants

### PR DESCRIPTION
**Pull request by airballking**
@moesenle I have repeatedly found myself needing these utility functions in my projects. Right now, I there only functions to look up the value constants knowing the keywords.

Do you think they could be useful to others?

### CODE REVIEW

**Comment by gaya-** [comment]
I'm not @moesenle but I'll still add my input if that interests anyone.
I like the idea of these utility functions.
Some comments for the implementation though:
* It would be nice if the functions were generic just like `symbol-code` (and maybe `symbol-codes`) so that `code-symbol` could also work on the `ros-message` itself and not only message type (see implementation of `symbol-code` if this was unclear).
* The implementation of `code-symbol` would be more efficient if it would utilize `rassoc` instead of first creating a new list with `remove` and then searching through that list again. Something similar to how `symbol-code` is doing the search right now. E.g.:

```lisp
(let ((pair (rassoc code (symbol-codes m))))
  (unless pair
    (error "Could not get symbol for code ~a of ROS message type ~a" code m))
  (car pair))
```

**Comment by airballking** [comment]
@gaya- Your code review is much appreciated!

* Cool that you also like these utilities. Then, I'd go ahead with them!
* I like the idea of improving the implementation of ```code-symbol``` by using ```rassoc```. I will give it a try!
* Having ```symbol-codes```, ```code-symbols```, and ```code-symbol``` also work on instances of a class is a good idea! I will look into it. However, it is not working for me with ```symbol-code```, either. I am using ```visualization_msgs``` as an example:

```lisp
ROSLISP> (symbol-code 'visualization_msgs-msg:marker :arrow)
0

ROSLISP> (symbol-code (make-msg "visualization_msgs/Marker") :arrow)
There is no applicable method for the generic function
  #<STANDARD-GENERIC-FUNCTION SYMBOL-CODE (2)>
when called with arguments
  (#<STANDARD-CLASS VISUALIZATION_MSGS-MSG:MARKER> :ARROW).
   [Condition of type SIMPLE-ERROR]
```

It seems to me, there is a bug. I'll correct it!

**Comment by airballking** [comment]
I screwed up the git history of in my fork branch! I'll have to switch to a new one...

**Comment by airballking** [comment]
OK, I have rewritten history in my fork to make a nice pull request.

@gaya- Could you please have another look at this PR?

File: src/msg.lisp [hunk 1]
```diff
@@ -75,9 +75,26 @@
...
 
+(defmethod code-symbols ((msg-type symbol) code)
+  (remove code (symbol-codes msg-type) :test-not #'= :key #'rest))
+
+(defmethod code-symbols ((m ros-message) code)
+  (code-symbols (type-of m) code))
+
+(defmethod code-symbol ((msg-type symbol) code)
+  (let ((pair (rassoc code (symbol-codes msg-type) :test #'=)))
+    (when pair
```
Comment (gaya-) [inline src/msg.lisp]: Could we unify the way corner cases are handled here, i.e. between `code-symbol` and `symbol-code`? Either get rid of errors in both or throw errors in both. Otherwise it might get confusing. I like it when errors are thrown in corner cases as it makes debugging much simpler and returning `NIL` is pretty generic and doesn't tell you much about what went wrong. But in this case it can very well be that one symbol or number is not defined in the message and I guess considering that an error is too harsh. However, I just grepped `symbol-code` in our code base and it's used all over the place. The errors are not handled anywhere but I'd guess, if this is such a widely-used function (and maybe not only by CRAM people) it might be too late to change its behaviour...

Comment (airballking) [inline src/msg.lisp]: We can add an error to ```code-symbol```, if you want.

Comment (gaya-) [inline src/msg.lisp]: (It's not like I want it, I'm only sharing my thoughts and suggestions. I'll try to phrase my sentences better next time.)
